File: in_Java/1190 Reverse Substrings Between Each Pair of Parentheses.java
class Solution {
    public String reverseParentheses(String s) {
        Stack<Character> st = new Stack<>();
        List<Character> temp = new ArrayList<>();
        for(char c : s.toCharArray()){
            if(c != ')') st.push(c);
            else{
                temp = new ArrayList<>();
                while(!st.isEmpty() && st.peek() != '('){
                    temp.add(st.pop());
                }
                st.pop();
                for(char x : temp) st.push(x);
            }
        }
        char[] r = new char[st.size()];
        int id = r.length-1;
        while(!st.isEmpty()) r[id--] = st.pop();
        return String.valueOf(r);
    }
}

File: in_Python/1190 Reverse Substrings Between Each Pair of Parentheses.py
class Solution(object):
    def reverseParentheses(self, s):
        """
        :type s: str
        :rtype: str
        """
        st = []
        for c in s:
            if c == '(':
                st.append(c)
            elif c != ')':
                st.append(c)
            else:
                temp = []
                while len(st) != 0 and st[-1] != '(':
                    temp.append(st.pop())
                if len(st) != 0:
                    st.pop()
                for t in temp:
                    st.append(t)
        res = ''
        for i in st:
            res += i
        return res
            
        